catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/Source.java

```diff
@@ -34,24 +34,42 @@
    *
    * <p><b>This is expected to be an expensive operation, possibly involving network I/O.</b>
    * Typically only {@link ddf.catalog.CatalogFramework} implementations will call this and only
-   * periodically.
+   * periodically. {@link
+   * ddf.catalog.CatalogFramework#getSourceInfo(ddf.catalog.operation.SourceInfoRequest)} should be
+   * used as a non-blocking alternative to {@link #isAvailable()}.
    *
    * @return true - if the site is available (up), false - if the site is unavailable (down)
-   * @see ddf.catalog.util.SourcePoller
    */
   public boolean isAvailable();
 
   /**
    * This method is the same as {@link #isAvailable()} but allows a caller to provide a {@link
-   * SourceMonitor} callback object which is meant to be used as a way for the {@link Source} to
-   * dynamically contact the caller with its availability. The intent is to return a boolean as to
+   * SourceMonitor} which is meant to be used as a way for the {@link Source} to dynamically update
+   * the caller when its availability changes. The intent is to return a {@code boolean} as to
    * whether the {@link Source} is available at this very moment and also to use the {@link
-   * SourceMonitor} object to update the caller of this method if this Source's availability changes
-   * later in the future.
+   * SourceMonitor} to update the caller if the availability of the {@link Source} is checked
+   * sometime in the future and it has changed. Implementations of this method should manage calling
+   * back to multiple {@link SourceMonitor}s.
    *
-   * @param callback - used to notify the caller of this method when the {@link Source} object wants
+   * <p>
+   *
+   * <p>NOTE: Consider using {@link
+   * ddf.catalog.CatalogFramework#getSourceInfo(ddf.catalog.operation.SourceInfoRequest)} instead of
+   * {@link #isAvailable()} for two reasons:
+   *
+   * <ol>
+   *   <li>The {@link ddf.catalog.CatalogFramework} is a common place where {@link Source}
+   *       availability is managed. Exceptions and timeouts can be handled consistently across all
+   *       {@link Source}s.
+   *   <li>The {@link ddf.catalog.CatalogFramework} periodically checks the availability of {@link
```

Comment:
✏️We may want to be a little more vague here and avoid putting constraints on the `CatalogFramework` interface (e.g., periodic availability checks) that would be easy to miss if that interface's description and its implementation ever changes in the future. Just recommending the use of `getSourceInfo()` might be good enough.

Reply:
```java
/**
 * This method is the same as {@link #isAvailable()} but allows a caller to provide a {@link
 * SourceMonitor} which is meant to be used as a way for the {@link Source} to dynamically update
 * the caller when its availability changes. The intent is to return a {@code boolean} as to
 * whether the {@link Source} is available at this very moment and also to use the {@link
 * SourceMonitor} to update the caller if the availability of the {@link Source} is checked
 * sometime in the future and it has changed. Implementations of this method should manage calling
 * back to multiple {@link SourceMonitor}s.
 *
 * <p>
 *
 * <p>NOTE: Consider using {@link
 * ddf.catalog.CatalogFramework#getSourceInfo(ddf.catalog.operation.SourceInfoRequest)} instead of
 * {@link #isAvailable()}.
 *
 * @param callback used to notify the caller of this method when the {@link Source} object wants
 *     to update its availability.
 * @return {@code true} - if the site is available (up), {@code false} - if the site is
 *     unavailable (down)
 */
public boolean isAvailable(SourceMonitor callback);
```